tests/contracts/test_derive_version.py

```
@@ -8,27 +8,44 @@
+
+
+def test_main_version_rejects_already_released_base_version(monkeypatch) -> None:
+    """Main prereleases should not continue after the target version has shipped."""
+
+    def fake_run_git(args: list[str]) -> str:
+        assert args == ["tag", "--merged", "HEAD", "--list"]
+        return "\n".join(["v0.2.2", "v0.3.0", "v0.3.1"])
+
+    monkeypatch.setattr(derive_version, "run_git", fake_run_git)
+
+    with pytest.raises(ValueError, match="Bump pyproject.version to the next release target"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)